src/mysql-util/src/schemas.rs

```diff
@@ -204,16 +221,26 @@ where
     Q: Queryable,
 {
     let table_rows: Vec<(String, String)> = match schema_request {
-        SchemaRequest::All => {
-            // Get all tables in non-system schemas.
-            // TODO(roshan): Many users create user-defined tables in the `mysql` system schema, since mysql doesn't
-            // prevent this. We may want to consider adding a warning for this in the docs, since that schema
-            // contains dozens of built-in system tables that we need to filter out.
-            let table_q = "SELECT table_name, table_schema
-                FROM information_schema.tables
-                WHERE table_type = 'BASE TABLE'
-                AND table_schema NOT IN ('information_schema', 'performance_schema', 'mysql', 'sys')";
-            conn.exec(table_q, ()).await?
+        SchemaRequest::All {
+            include_system_schemas,
+        } => {
+            if *include_system_schemas {
```

Comment:
This control flow really looks like it wants to become its own separate enum variant, there is no shared logic between the two cases. I think it will be a bit cleaner if we had `All` and `AllWithSystemSchemas` or something along those lines.

Reply:
Ah yeah that would be cleaner - will restructure!